keps/sig-auth/20190607-certificates-api.md

```diff
@@ -292,8 +293,30 @@ type CertificateSigningRequestStatus struct {
   // Conditions applied to the request, such as approval or denial.
   Conditions []CertificateSigningRequestCondition
 
-  // If request was approved, the controller will place the issued
-  // certificate here.
+  // Certificate is populated by the signer with the issued certificate in PEM format.
+  // In JSON and YAML output, this entire field is base64-encoded, so it consists of:
+  // base64(
+  // -----BEGIN CERTIFICATE-----
+  // MII...Pb7Yu/E=
+  // -----END CERTIFICATE-----
+  // optional intermediate certificate blocks
+  // -----BEGIN CERTIFICATE-----
+  // MII...MGKB
+  // -----END CERTIFICATE-----
+  // -----BEGIN CERTIFICATE-----
+  // MII...AY1M
+  // -----END CERTIFICATE-----
+  // )
+  //
+  // In v1beta1, this field is unvalidated.
+  //
+  // In v1, this field is validated to contain one or more PEM blocks.
+  // All PEM blocks must have the "CERTIFICATE" label, contain no headers, and the encoded data
+  // must be a BER-encoded ASN.1 Certificate structure as described in Section 4 of RFC5280.
+  // If more than one PEM block is present, and the definition of the requested spec.signerName
+  // does not indicate otherwise, the first block is the issued certificate,
+  // and subsequent blocks should be treated as intermediate certificates and presented in TLS handshakes.
+  // Non-PEM content may appear in between the CERTIFICATE PEM blocks and is unvalidated.
```

Comment:
Is this to facilitate things like comments?

Reply:
```
5.2.  Explanatory Text

   Many tools are known to emit explanatory text before the BEGIN and
   after the END lines for PKIX certificates, more than any other type.
   If emitted, such text SHOULD be related to the certificate, such as
   providing a textual representation of key data elements in the
   certificate.

Subject: CN=Atlantis
Issuer: CN=Atlantis
Validity: from 7/9/2012 3:10:38 AM UTC to 7/9/2013 3:10:37 AM UTC
-----BEGIN CERTIFICATE-----
MIIBmTCCAUegAwIBAgIBKjAJBgUrDgMCHQUAMBMxETAPBgNVBAMTCEF0bGFudGlz
MB4XDTEyMDcwOTAzMTAzOFoXDTEzMDcwOTAzMTAzN1owEzERMA8GA1UEAxMIQXRs
YW50aXMwXDANBgkqhkiG9w0BAQEFAANLADBIAkEAu+BXo+miabDIHHx+yquqzqNh
Ryn/XtkJIIHVcYtHvIX+S1x5ErgMoHehycpoxbErZmVR4GCq1S2diNmRFZCRtQID
AQABo4GJMIGGMAwGA1UdEwEB/wQCMAAwIAYDVR0EAQH/BBYwFDAOMAwGCisGAQQB
gjcCARUDAgeAMB0GA1UdJQQWMBQGCCsGAQUFBwMCBggrBgEFBQcDAzA1BgNVHQEE
LjAsgBA0jOnSSuIHYmnVryHAdywMoRUwEzERMA8GA1UEAxMIQXRsYW50aXOCASow
CQYFKw4DAh0FAANBAKi6HRBaNEL5R0n56nvfclQNaXiDT174uf+lojzA4lhVInc0
ILwpnZ1izL4MlI9eCSHhVQBHEp2uQdXJB+d5Byg=
-----END CERTIFICATE-----

            Figure 7: Certificate Example with Explanatory Text
```

https://tools.ietf.org/html/rfc7468#section-5.2

Reply:
it just allows for signers that produce a human readable version outside the PEM block, e.g.:

```
Certificate:
    Data:
        Version: 3 (0x2)
        Serial Number:
            31:33:63:24:cc:c4:dd:e7:0b:a0:7c:d9:69:df:a2:94:ee:58:2c:5f
    Signature Algorithm: sha256WithRSAEncryption
...
-----BEGIN CERTIFICATE-----
MIIDh...RMX74c=
-----END CERTIFICATE-----
```